Expensify/App
# Implement Archived Workspace Chat UI

### Details
Implements the Archived Workspace Chat UI.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/195474

### Tests
#### Storybook
1. Run storybook, verify that you can see the new stories for the `Banner` component.

    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159600141-13b78ae1-2371-4ff9-8a4c-d6bcdfc863f0.png">
    
    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159600345-89ba0048-806a-40ba-9f63-8d5f9600111c.png">
    
    There are a number of issues with Storybook at present, none having to do with this PR or the new stories.

#### Setup
1. Create and validate 5 accounts – 1 policy admin and 4 employees.
1. Sign in as the policy admin in NewDot.
1. Create a new workspace.
1. Go to `Settings` > `<Your workspace>` > `Manage members` and invite all four employees to the workspace.

#### Account Closed

1. Open the workspace chat for employee A. Verify that it's in the open state.

    - `BUG`: I had to sign out and sign back in for the workspace chat(s) to appear.

        <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159602911-b2a402d6-0500-4049-8ef6-ca1ee01db619.png">
    
1. Sign into OldDot as employee A. Go to `Settings` > `Account` > `Account Details` and close the account.

    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159603498-17f20dc7-1d71-4b98-b8b0-548ad8441223.png">

1. Go back to NewDot. Verify that the workspace chat for employee is archived, and that you see the following:

    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159604121-9cbf17aa-eb88-4768-8041-560662cd9df8.png">
    
    - `BUG`: I had to refresh the page for the UI to update.

1. Go back to the workspace chat for the employee w/ the closed account, and verify that you see the following:

     <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159604639-62e43d6d-1728-4f20-9555-02c4c4bab314.png">

#### Account Merged
1. Sign in to NewDot as employee B.
1. Go to `Settings` > `Profile` and set a first and last name
1. Sign in to NewDot as employee C.
1. Go to `Settings` > `Profile` and set a first and last name
1. Sign in to NewDot as the admin and find the workspace chat for employee C.

    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159617862-91672a17-6e12-4fd5-a3c2-9488a819f197.png">

1. Sign in to OldDot as employee B.
1. Go to `Settings` > `Account` > `Account Details` and merge employee C's account into employee B's account.
    
    <img width="1101" alt="image" src="https://user-images.githubusercontent.com/47436092/159617928-9b88f080-2bce-4a51-a53c-094dbb1373b2.png">

1. Go back to NewDot and employee C's workspace chat should now look like this:

    <img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159622550-ff4b7518-66e8-4323-99bc-64957e9180a2.png">

1. Go to NewDot `Settings` > `Preferences` and switch the language to Spanish.
1. Go back to the workspace chat for employee C and you should see the following copy:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/47436092/159622817-52759094-147b-4092-bb4f-17b773fb1bf1.png">
     
#### Removed from Policy
Testing blocked on https://github.com/Expensify/Auth/pull/6456

#### Policy Deleted
Testing blocked on https://github.com/Expensify/Auth/pull/6456


### QA Steps
Most of this will be tested internally before we add a large suite of regressions to cover workspace chats. However, there is something that needs to be QA'd here (Chrome web only) to prevent performance regressions.

1. Sign out.
1. Open the JS console. Make sure you have verbose logs enabled. This dropdown should say `All Levels`:

    <img width="120" alt="image" src="https://user-images.githubusercontent.com/47436092/159772527-aef622ac-d732-4157-8ff3-037571c086aa.png">

1. Sign into a high traffic account with very large policies.
1. Search the logs for `Timing:expensify.cash.personal_details_formatted`
1. Verify that the number is very low (ideally less than `1000ms`).
1. Repeat these same steps on production with the same account, and post the difference in timing between staging and production in this PR.

### Tested On


### Screenshots


#### Web
All of the screenshots for the above steps were taken on web.

#### Mobile Web
<img width="350" alt="image" src="https://user-images.githubusercontent.com/47436092/159604992-57bd0804-2574-4bf2-80df-d35c7bef3923.png">

<img width="350" alt="image" src="https://user-images.githubusercontent.com/47436092/159624388-6d37159e-6c82-45a7-a94d-fc62bc3679fb.png">

#### Desktop
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/47436092/159627266-09a6a99d-47e7-4918-a297-b625d370948e.png">

<img width="1203" alt="image" src="https://user-images.githubusercontent.com/47436092/159627290-215ff20d-c505-460e-a2f5-ee74c0fddfdf.png">

#### iOS
<img width="350" alt="image" src="https://user-images.githubusercontent.com/47436092/159625618-db0f43c5-fac3-4ef2-8839-77c6b097162f.png">

<img width="350" alt="image" src="https://user-images.githubusercontent.com/47436092/159625659-18b90991-85c9-4553-928e-cde9256a52a5.png">

#### Android
<img width="350" alt="" src="https://user-images.githubusercontent.com/47436092/159626911-92c51f36-903a-4c97-abb4-1dc3c3475eaa.png">

<img width="350" alt="" src="https://user-images.githubusercontent.com/47436092/159626961-9451d3c6-608d-43e3-87a7-3551a68d0b0b.png">
